lib/Monads/reader_option/Reader_Option_VCG.thy

```diff
@@ -124,6 +134,16 @@ lemma owhile_ovalid[wp]:
    apply auto
   done
 
+lemma case_options_weak_ovalid:
+  "\<lbrakk> \<lblot>P\<rblot> f \<lblot>Q\<rblot>; \<And>x. \<lblot>P'\<rblot> g x \<lblot>Q\<rblot> \<rbrakk>
+   \<Longrightarrow> \<lblot>P and P'\<rblot> case opt of None \<Rightarrow> f | Some x \<Rightarrow> g x \<lblot>Q\<rblot>"
```

Comment:
Curious that we need this one. Does `wpc` work for `ovalid`? Do we still need to set that one up?

Reply:
I'm not sure I've tried `wpc` with `ovalid`. Actually, this lemma ended up not being used in the work I thought I wanted it for. Should I just delete it?

Reply:
Yes, let's scrap it. If `wpc` isn't set up yet, we should probably do that in this theory, but some other time.

Reply:
Great. Should I squash and force push so that you can do one last check of the commits?

Reply:
yes, please